java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowConfigBuilder.java

```diff
@@ -216,7 +216,7 @@ public JdbcToArrowConfigBuilder setTargetBatchSize(int targetBatchSize) {
    * Defaults to wrapping {@link JdbcToArrowUtils#getArrowTypeFromJdbcType(JdbcFieldInfo, Calendar)}.
    */
   public JdbcToArrowConfigBuilder setJdbcToArrowTypeConverter(
```

Comment:
Is it possible to add an overload to accept a `Function` still for backwards compatibility?

Reply:
Alternatively - we could add the column index as a field in `JdbcFieldInfo` so that signatures do not have to change

Reply:
Thank you for review @lidavidm 
JdbcFieldInfo with additional column index is works for me